### ISSUE TITLE
bug: Java17 lambda function fails with  READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE: java.lang.NoSuchFieldError

### ISSUE THREAD

**Issue by nirupamadsk** [localstack/localstack#9372]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

With AWS java libs upgrade to 1.1.3 we have started seeing this issue https://github.com/aws/aws-lambda-java-libs/issues/448

We are using localstack 2.3.3.dev and finding the same error while running the lambda. Going back to localstack 2.2.dev didn't help here. 

Is it happening because localstack always pull the latest  public.ecr.aws/lambda/java:17? Are there any workarounds with localstack currently so that we can run lambda with aws-lambda-java-libs-1.1.2

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

1. Compile a simple lambda handler with jackson 2.15.2 which tries to use one of the removed Jackson deserialization properties.

```
import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.fasterxml.jackson.annotation.JsonFormat.Feature;
import com.fasterxml.jackson.databind.MapperFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.json.JsonMapper;

public class HelloWorldHandler implements RequestHandler<Object, String> {

  public static final ObjectMapper OBJECT_MAPPER = JsonMapper.builder()
    .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
    .build();

  @Override
  public String handleRequest(Object input, Context context) {
    System.out.println(Feature.READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE);
    return "Hello World!";
  }
}
```

Step 2. Just try to deploy a java17 lambda function with the handler above. You should see the lambda execution failure. 

Step 3. Notice that in the lambda container you will find this aws library which comes with jackson-databind-2.15.2 but doesn't have jackson-annotations. This is a bug in aws lib as mentioned above, which they will fix soon. 

/var/runtime/lib/aws-lambda-java-serialization-1.1.3.jar

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
```
  localstack:
    container_name: "fdxs-localstack"
    image: "localstack/localstack:latest"
    ports:
      - "4566:4566" # LocalStack Gateway
      - "4510-4559:4510-4559" # External services port range
      - "7050:7050" # Lambda debug
    env_file:
      - .env
    environment:
      - MAIN_CONTAINER_NAME=fdxs-localstack
      - LS_SERVICES=iam,cloudwatch,dynamodb,s3,sns,sts,kinesis,kms
      - LS_LOG=warn
      - DEBUG=1
      - AWS_ACCESS_KEY_ID=test
      - AWS_SECRET_ACCESS_KEY=test
      - AWS_DEFAULT_REGION=us-east-1
      - LAMBDA_RUNTIME_ENVIRONMENT_TIMEOUT=60
      - GODEBUG="asyncpreemptoff=1"
    volumes:
      - ".:/fdxs"
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "${LOCALSTACK_VOLUME_DIR:-/var/lib/localstack}:/var/lib/localstack"
      - "./runlocal/localstack/scripts/init-aws.sh:/etc/localstack/init/ready.d/init-aws.sh"
    healthcheck:
      test: /fdxs/runlocal/healthcheck.sh http://localstack:4566/_localstack/init/ready 'SUCCESSFUL' 12 5
      interval: 30s
      timeout: 50s
      start_period: 40s
      retries: 1
    restart: always
    networks:
      - "local"
```

### Environment

```markdown
- OS:
- LocalStack:
```

**Comment by nirupamadsk** [comment]
@simonrw I've updated the issue with more details. Please check. I'll be happy to hop on a meeting with you if that makes things easier to reproduce.

**Comment by anton-shchyrov** [comment]
@simonrw This is a bug in the Amazon public.ecr.aws/lambda/java:17 container. This error also started appearing for me a week ago, although the code did not change. I fixed it by specifying an older version of the container 'public.ecr.aws/lambda/java:17.2023.07.19.04'

**Comment by nirupamadsk** [comment]
@anton-shchyrov where did you specify the older version?

**Comment by anton-shchyrov** [comment]
@nirupamadsk In dockerfile
```
FROM public.ecr.aws/lambda/java:17.2023.07.19.04
```

**Comment by nirupamadsk** [comment]
@anton-shchyrov I'm using localstack which internally pulls or comes with the lambda image perhaps, so I don't have control over the lambda image. That's why I opened this issue to know how to trick localstack to use a specific lambda17 image.

**Comment by joe4dev** [comment]
@anton-shchyrov is correct. This is a bug in the official AWS Lambda base image `amazon/aws-lambda-java:17.2023.10.16.09` reported here https://github.com/aws/aws-lambda-base-images/issues/117

---

👉 @nirupamadsk As a temporary workaround, you can use the [LocalStack configuration](https://docs.localstack.cloud/references/configuration/#lambda) `LAMBDA_RUNTIME_IMAGE_MAPPING` to use the previous version of the image. For example:

```python
LAMBDA_RUNTIME_IMAGE_MAPPING={"java17": "public.ecr.aws/lambda/java:17.2023.07.19.04"}
````

**Comment by aleskiewicz** [comment]
This was an issue intoduced in `aws-lambda-java-serialization:1.1.3` https://github.com/aws/aws-lambda-java-libs/issues/448  It was just resolved and released in version `1.1.4`.  Unfortunately today's base image release does not contain the fix yet (`public.ecr.aws/lambda/java:17.2023.10.19.15`)

**Comment by joe4dev** [comment]
> This was an issue intoduced in `aws-lambda-java-serialization:1.1.3` [aws/aws-lambda-java-libs#448](https://github.com/aws/aws-lambda-java-libs/issues/448) It was just resolved and released in version `1.1.4`. Unfortunately today's base image release does not contain the fix yet (`public.ecr.aws/lambda/java:17.2023.10.19.15`)

I see. Thank you for the update @aleskiewicz 

Let's hope the AWS team releases an updated base image soon. There's not much we can do beyond offering a configuration option for an alternative base image. Hence, feel free to create a custom image with the fixed version `1.1.4` and configure LocalStack via the `LAMBDA_RUNTIME_IMAGE_MAPPING` to use that image. Is that a viable workaround?

**Comment by nirupamadsk** [comment]
Thanks @joe4dev. While we wait for AWS to publish the latest image with the fix, the suggestion on how to use a specific version of lambda17 worked for us.
